File: src-tauri/src/models/competitor.rs
use serde::{Deserialize, Serialize};
use specta::Type;

#[derive(Serialize, Deserialize, Type, Debug, Clone)]
#[serde(rename_all = "camelCase")]
pub struct Competitor {
    pub id: String,
    pub first_name: String,
    pub last_name: String,
    pub birth_date: String,
    pub gender: String,
    pub club: Option<String>,
    pub city: Option<String>,
    pub notes: Option<String>,
    pub competition_order: i64,
    pub photo_base64: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize, Type, Debug, Clone)]
#[serde(rename_all = "camelCase")]
pub struct CompetitorCreate {
    pub first_name: String,
    pub last_name: String,
    pub birth_date: String,
    pub gender: String,
    pub photo_base64: Option<String>,
    pub photo_filename: Option<String>,
}

File: werewolf-cloud/packages/domain/src/models/competitor.ts
import { z } from 'zod';

export const genderSchema = z.enum(['Male', 'Female']);
export type Gender = z.infer<typeof genderSchema>;

export const competitorSchema = z.object({
  id: z.string().uuid(),
  firstName: z.string().min(1),
  lastName: z.string().min(1),
  birthDate: z.string().regex(/\d{4}-\d{2}-\d{2}/, 'birthDate must be YYYY-MM-DD'),
  gender: genderSchema,
  club: z.string().nullable().optional(),
  city: z.string().nullable().optional(),
  notes: z.string().nullable().optional(),
  photoData: z.instanceof(ArrayBuffer).nullable().optional(),
  photoFormat: z.string().nullable().optional(),
  photoMetadata: z.string().nullable().optional(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
});

export type Competitor = z.infer<typeof competitorSchema>;

export const competitorCreateSchema = competitorSchema.pick({
  firstName: true,
  lastName: true,
  birthDate: true,
  gender: true,
  club: true,
  city: true,
  notes: true,
});

export type CompetitorCreateInput = z.infer<typeof competitorCreateSchema>;

export const competitorUpdateSchema = competitorCreateSchema.partial();

export type CompetitorUpdateInput = z.infer<typeof competitorUpdateSchema>;
